### PR TITLE
Docs Update

### PR DESCRIPTION
**Pull request by Tetsuro**
# Docs Update

This PR updates the styles and content of the Liquid docs. 

To run locally: 

1. Navigate to the `liquid` folder 
2. Install dependencies by running `npm install`
3. Run `grunt` (make sure you have [grunt-cli](http://gruntjs.com/getting-started#installing-the-cli) installed!)
4. Visit `localhost:4000`.

### CODE REVIEW

**Comment by parkr** [comment]
DOPE!

**Comment by parkr** [comment]
Let me know how I can help.

**Comment by Tetsuro** [comment]
Woo! Will keep you posted @parkr :)

**Comment by Tetsuro** [comment]
Hey @parkr I added a README here: https://github.com/Shopify/liquid/tree/docs-update for when you'd like to preview the site. I have a Jekyll site setup now with some basic styling.

**Comment by Tetsuro** [comment]
ping @parkr :)

**Comment by parkr** [comment]
Liquid is built for getting things done, and it's pretty great at that. I think the docs should pertain to use-cases, rather than the discrete function of each tag or filter. I'm not sure how I'd express those use-cases, but I think I'd try to be as non-technical as possible. The shopify liquid docs are a bit opaque at the moment :/

**Comment by Tetsuro** [comment]
@parkr Hmm I see. Have you got any examples of other documentation sites that you like? I still think it'd be a good idea to have a quick reference of all filters and tags, but maybe I'm in the minority. 

In the meantime, I'm going to focus on setting up the Jekyll site so that we can toss in the content as soon as we have that sorted out.

**Comment by jglovier** [comment]
@Tetsuro let me know if I can help anything (styling especially).

**Comment by parkr** [comment]
How do you feel about a design like this? http://tomjohnson1492.github.io/jekyll-doc/

**Comment by Tetsuro** [comment]
I think we should really nail down the structure of the content first before we move any further with the design. The way we decide to categorize the information for Liquid will have an impact on the design. So I guess my question is, what content do we have to work with now and how do we want to break them up into categories? 

@fw42, would you still want to go with the approach of porting over Shopify's Liquid Docs and then removing and Shopify-specific tags? This would be the quickest way to do this. 

IMHO @parkr, I'm not too keen on the design of that site... sorry! :sob:  I'm a big fan of simple designs like the [Frame Docs](http://framerjs.com/docs/).

**Comment by Tetsuro** [comment]
Thanks @jglovier, will let you know :)

**Comment by fw42** [comment]
@Tetsuro: I think we should go with whatever is the easiest to maintain and the easiest to keep in sync with the Shopify docs, since there is a huge overlap and realistically, nobody will have time to maintain two docs that are totally different, imho.

@dtKinger: what do you think?

**Comment by parkr** [comment]
Can we set this up so these docs and the shopify docs use a submodule to keep all the overlapping info in sync? It's not ideal (you can't update docs in a PR with the updated code, for example) but maybe it'll work best?

Why have 2 different sites? Have a big flag at the top of Shopify docs (not usage guides) that says "THIS IS A SHOPIFY-ONLY FILTER/TAG" or something

**Comment by Tetsuro** [comment]
@fw42 - When you have a moment, do you think I could get your help with getting some of the custom Liquid helpers? Ex: `{% table_of_contents %}` and `{% anchor %}`.

**Comment by parkr** [comment]
Any movement here?

**Comment by Tetsuro** [comment]
Hey @parkr! Last week I spent some time setting up Grunt for our docs so that we can add neat things like auto-prefixing. I also spent some time back in Sketch to try and find a way to improve the sidebar design. 

I'd like to spend some time this weekend on the sidebar design + implementation.

**Comment by parkr** [comment]
Hey @Tetsuro! Thanks for getting back to me. Sounds great to me. I look forward to seeing the new/improved designs!!!!

**Comment by Tetsuro** [comment]
Hey @parkr! Got some work done on the sidebar last night! 

- Sidebar now looks for pages from folders in the root level of the site, defined [here](https://github.com/Shopify/liquid/commit/65311ca5478d31d27708149e72d1d8b32eca3618#diff-2c19d9859b055d0302043d0fa2833e3fR7). Currently it's set up so that it pulls in pages from three folders: `filters`, `objects`, and `tags`, but please feel free to change the structure here. I just wanted to have folders in there to get the foundation set up. 
- Active section expands, non-active sections collapse. Active page also highlighted. 
- [Major refactoring of SCSS](https://github.com/Shopify/liquid/commit/b391d97ddba1f1aaf0b26b6ab908a66183e07a68). 

Next, I'll be tackling: 
- Sexier sidebar scrollbar (maybe use a custom plugin).
- Better homepage

**Comment by arthurnn** [comment]
Not sure if I asked this already, but what would be the strategy to deploy this?
IMO we should have another repo that only have the liquid docs and we merge there the compiled version of the html and the dev ones. thoughts?
cc @fw42

**Comment by fw42** [comment]
:-1: to having another repo.. docs should be in this repo, imho. I would say the source for the docs should be on master and a compiled version on the gh-pages branch. That way code changes and docs changes can be in the same PR. (Didn't we have this discussion already?)

**Comment by fw42** [comment]
what's with all the node_modules files in this branch? do we really need to commit that stuff?

**Comment by pushrax** [comment]
node_modules should be ignored.

**Comment by arthurnn** [comment]
@fw42 we had that discussion but I could not find it so i wanted to double check.
I will try to make that happen, and cleanup and merge this.. We need to get this merged to master, and we can iterate on top of this ground work.

**Comment by Tetsuro** [comment]
@pushrax My bad on that! I think I've fixed it up so that the `node_modules` are not included. 

@parkr When you have a second, let me know what you think of the sidebar. Let's start adding more content :)

**Comment by parkr** [comment]
So awesome! Why not use collections? I think it'd be cleaner to reason about. You can output collection documents into their own files, so it suits your wish exactly.

**Comment by Tetsuro** [comment]
@parkr Interesting! I didn't know about collections tbh :P, I will look into that!

**Comment by parkr** [comment]
Where are we on this?

**Comment by Tetsuro** [comment]
@parkr Just starting with the conversion to Collections. Still aiming for the end of July launch date.

**Comment by parkr** [comment]
Sounds brilliant!!

**Comment by Tetsuro** [comment]
@parkr I have a noob question when you have a moment :( 

I've got a collection called `tags`, but this is problematic when I try to access the pages in it because `site.tags` is already a variable in Jekyll. So when I try to loop through `site.tags` or `site.["tags"]` it'll loop through the site's tags instead of the pages in the `tags` collection. Do you know if there's a way around this?

**Comment by parkr** [comment]
> I've got a collection called tags, but this is problematic when I try to access the pages in it because site.tags is already a variable in Jekyll. So when I try to loop through site.tags or site.["tags"] it'll loop through the site's tags instead of the pages in the tags collection. Do you know if there's a way around this?

Try `site.collections["tags"]`. It's a collection document. You can iterate over `site.collections["tags"].docs` or any other variable from here: https://github.com/jekyll/jekyll/blob/master/lib/jekyll/collection.rb#L151-L158

**Comment by Tetsuro** [comment]
@parkr Ahhh gosh, so obvious :sweat_smile: Thanks Parker! Follow-up question: do you know if `page.next` and `page.previous` works for collection pages? I can't seem get it to output anything :( 

Updated sidebar to use collections, refactored a bunch of CSS/HTML. I also made it so that the sidebar sections no longer collapse - I think it's better this way as it lets the viewer skim through the available tags/fitlers/etc quicker.

Currently working on moving over more content!

**Comment by Tetsuro** [comment]
Also, I'm thinking the "Objects" section isn't really necessary. I brought it over initially because I was using the Shopify Liquid Docs as a reference, but everything there is pretty much Shopify-specific. I think it'd be better to bring in some of the [Basics](https://docs.shopify.com/themes/liquid-documentation/basics) pages from there instead. Let me know what you think!

**Comment by parkr** [comment]
> I'm thinking the "Objects" section isn't really necessary.

Agreed! That would be Jekyll/Shopify-specific, so that can be omitted. I love the Basics page. Could that be the landing page for the website? The first thing users see? Would be nice to have that kind of gentle overview spoon-fed to the user.

> do you know if page.next and page.previous works for collection pages?

Not yet :( Want to add it? [Posts do it like this.](https://github.com/jekyll/jekyll/blob/master/lib/jekyll/post.rb#L294-L310) Not sure how efficient that is – maybe there's room for improvement.

**Comment by Tetsuro** [comment]
> Agreed! That would be Jekyll/Shopify-specific, so that can be omitted. I love the Basics page. Could that be the landing page for the website? The first thing users see? Would be nice to have that kind of gentle overview spoon-fed to the user.

That sounds good - I can move that to the front page :)

> Not yet :( Want to add it? Posts do it like this. Not sure how efficient that is – maybe there's room for improvement.

Actually may not need pagination after all. I'm thinking of just outputting every markdown file's contents onto a single page so that it's easier to see a section at a glance.

**Comment by Tetsuro** [comment]
heya @parkr, I was hoping you could help me out with this thing I'm stuck with...

One of the feedbacks that I got from a coworker was that instead of having the contents of all pages in one page, so that you can easily scroll through each page (as opposed to having to click each one). For example, all filters will be output in `/filters/index.html`. 

What I can't figure out is how to ignore `/filters/index.html` from the loop that output the content [here](https://github.com/Shopify/liquid/commit/d2a5550a5ed7e2e5aa6e66c85b429dd995bde156#diff-2c19d9859b055d0302043d0fa2833e3fR31). I'm doing a ghetto `contains` here for now, but if I do that, it throws off the content of the remaining markdown files because `index.html` is an HTML file. Here's a screenshot of what I mean:

screen shot 2015-08-15 at 7 12 01 pm

TLDR: How can I have an index page for each collection (Filters, Basics, Tags) that outputs the content of all markdown files within it?

Thanks! Hope that makes sense, please let me know if anything's unclear.

**Comment by parkr** [comment]
You'll want to use a normal page for that instead of a collection doc. Throw it in filters/  in your source and you should be ok.

**Comment by Tetsuro** [comment]
@parkr jeez of course! That makes sense... I've fixed up the navigation and made it so that the contents of each collection is displayed on one page, and the links in the sidebar are deeplinks in their respective page. Have a look and let me know what you think! 

Missed my self-imposed deadline by a mile :( I started a [checklist](https://github.com/Shopify/liquid/pull/513#issue-55549866) to keep track of what else needs to be done. I think most of it has to do with the content side of things (which I could totally use a hand with), in particular the Basics collection. I think the site itself is in pretty good shape :+1:

**Comment by parkr** [comment]
Hm...

> Have a look and let me know what you think!

Hm... how do I build it? I ran `jekyll serve` but the CSS file linked to in the HTML wasn't there. Any thoughts?

> I think most of it has to do with the content side of things

Out of curiousity, what can be shipped right now and be added later?

**Comment by Tetsuro** [comment]
> Hm... how do I build it? I ran jekyll serve but the CSS file linked to in the HTML wasn't there. Any thoughts?

Ah! This is because the compiled site (the `_site` folder) is not part of the repo. Apologies for the horrible instructions! I've updated the [top comment](https://github.com/Shopify/liquid/pull/513#issue-55549866) with steps on how to run the site using Grunt. Please let me know if you run into any issues with this. 

> Out of curiousity, what can be shipped right now and be added later?

That's a good question. I don't think we're ready to ship just yet. The first priority would be to go through the Basics pages, reorganize some of the content there, making sure everything there is relevant. This is next on my list, and it'd be sweet to have you proof-read the content afterwards. 

After that I'm going to go through the FED tasks and make sure the site looks good + add final polishes. 

Once we get the above done I think we'd in good shape to ship. We can add the content under "Missing content" above post-shipping.

**Comment by parkr** [comment]
Sweet! Let me know when and what for the proof-read. :)

Thanks for the compilation instructions. Are you still planning on shipping on GitHub Pages? If so (and I think it'd be the best choice), maybe caching the compiled CSS will be best.

**Comment by Tetsuro** [comment]
Hey @parkr! Were you able to compile and run Grunt alright? :)

I worked on this a bit more today, putting in more content for "Basics" and "Tags". I'm going through each page now, removing any Shopify-specific content and fixing up the formatting. 

It'd be sweet to have your :eyes: on the Basics section! I'm having a bit of a hard time figuring out what the best way to organize all the content under that section is... right now I have all of the content being output on one page (like I'm doing with the Filters section), but I'm starting to think that's a bad idea because each page is pretty heavy on content. Would love to hear your thoughts on that! I can totally revert it so that each link in the sidebar links to its own individual page, if need be.

> Are you still planning on shipping on GitHub Pages? If so (and I think it'd be the best choice), maybe caching the compiled CSS will be best.

That sounds great! Though I'd like to double-check with @fw42 on that one. 

Really appreciate your patience with this project. Hoping to get this out the door ASAP!

**Comment by parkr** [comment]
> It'd be sweet to have your  on the Basics section!

Let me know when it's ready – just looking at it now and it's all very Shopify-specific.

> I'm having a bit of a hard time figuring out what the best way to organize all the content under that section is... right now I have all of the content being output on one page (like I'm doing with the Filters section), but I'm starting to think that's a bad idea because each page is pretty heavy on content. Would love to hear your thoughts on that!

I think Basics should be a terse section, so I'd recommend keeping it on one page. If it feels too big, see if anything can be trimmed down. If I were to sit down with someone unfamiliar to Liquid and run through the "basics," I'd do a walkthrough:

1. Start with what it does (and maybe 1 or 2 things it might be mistaken for being able to do). "Liquid is a library and a language for writing and rendering templates. Templates allow one to reduce duplication and change the output without having to change the code itself." (Badly put, but you know)
2. Show crazy basic example – output a variable
3. Go through basic example and describe `{{`, the variable, and `}}`.
4. Show basic example – write an `if` statement
5. Go through the `if` statement token by token. `{% if %}`, the conditional statement, `{% endif %}`
6. Show more complicated example – `for` loop
7. Go through `for` loop token by token.

And therein, link to all possibilities, like in the conditional, have a link to special keywords like `!= empty` or something.

Also, in Jekyll 3 beta, the iteration over collections changes a bit. Just a head's up :)

**Comment by Tetsuro** [comment]
@parkr Appreciate the feedback! You're right, the basics should be as concise as possible. 

I do however think there is a need for more detailed pages for things such as "Truthy and Falsy" or "Operators"... so maybe those can live in their own individual pages, and the super-terse "basics" can just be part of the homepage. 

> Also, in Jekyll 3 beta, the iteration over collections changes a bit. Just a head's up :)

Will keep that in mind!

**Comment by parkr** [comment]
> I do however think there is a need for more detailed pages for things such as "Truthy and Falsy" or "Operators"

Definitely a great idea to delve into these topics more in-depth. Agreed that they deserve their own pages.

> the super-terse "basics" can just be part of the homepage.

I'd have a separate "Walkthrough" page or something similar. A "crash course," if you will. We do a bit of this in the Jekyll docs.

**Comment by parkr** [comment]
Anything else you need here?

**Comment by Tetsuro** [comment]
Just some more time to write out the content for "Basics" and "Walkthrough" sections... weekends just aren't long enough :sweat_smile:

**Comment by parkr** [comment]
@Tetsuro Ok! Is no time given during the work week? Might be nice to ask for a day to do this stuff and crank it out if that's kosher at Shopify. :+1:

**Comment by Tetsuro** [comment]
@parkr Absolutely agree! There is no time given during the week, but there is Hack Days coming up and I'm planning to focus solely on these docs. Hoping to get a lot done then!

**Comment by parkr** [comment]
HURRAH! Let me know how I can help further.

**Comment by Tetsuro** [comment]
hey @parkr! I've got a question regarding folders inside collections. 

Let say I have a collection folder called `_tags`, and I have this kind of structure:

```
_tags
  |--  control-flow
    |-- if.md
    |-- when.md
    |-- case.md
  |--  iteration
    |-- for.md
    |-- unless.md
  |--  variable
    |-- assign.md
    |-- capture.md
```

I then want a page at `/tags/control-flow` to concat all the markdown files in the `control-flow` folder and show them on one page (like I'm doing with Filters currently), and the same for `iteration` and `variable` with their respective markdown files. How can I iterate through folders inside a collection to do this? Or is there a better way to do this?

Thanks!

**Comment by parkr** [comment]
As of right now, this is only possible by checking the path:

```liquid
{% for tag in site.tags %}
  {% if tag.path contains "/iteration" %}
    <!-- Print -->
  {% endif %}
{% endfor %}
```

In the final version of Jekyll 3, this should be automatically categorized, i.e. the `categories` property would contain a list of its subdirectories.

**Comment by parkr** [comment]
@Tetsuro Did that answer your question?

**Comment by admhlt** [comment]
Thanks Parker. By "final version of Jekyll 3", do you mean 3.0, which is released now?

I'll let Tets respond to your question since I'm not certain what he's trying to do. :)

**Comment by parkr** [comment]
> > In the final version of Jekyll 3, this should be automatically categorized, i.e. the categories property would contain a list of its subdirectories.
>
> By "final version of Jekyll 3", do you mean 3.0, which is released now?

As it turns out, this feature was _not_ added. Posts became a collection, and subdirectories were always allowed in posts _without_ adding categories, so we couldn't add this feature in. If we allow parent directories of collections in the future, then those will be added to categories (i.e. `control-flow/_tags` would work, but not `_tags/control-flow`).

> I then want a page at /tags/control-flow to concat all the markdown files in the control-flow folder and show them on one page (like I'm doing with Filters currently), and the same for iteration and variable with their respective markdown files. How can I iterate through folders inside a collection to do this? Or is there a better way to do this?

The best way to do this is to have an `index.md` file inside each subdirectory which iterates over the collection documents and asks which contains that subdirectory in its path.

**Comment by parkr** [comment]
Every time I go to http://liquidmarkup.org/ to look up a filter or something, I become very sad.

@Tetsuro Need any more assistance here?

**Comment by Tetsuro** [comment]
@parkr, I know that feel :( It makes me a sad panda too. 

I was away on vacation for a bit and getting back into the swing of things. I'll need some more time with @admhlt again to jam out on some content though.

@admhlt - maybe we can take some time on Friday afternoon to work on this? Or at least do an audit of what's left? It'd be ridiculous to save this for another Hack Days, haha.

**Comment by admhlt** [comment]
Bad news! I'm out of town from Friday morning until Sunday.

Next week. Let's do this. For realsies. We could stay late after work and drink beers or something.

**Comment by parkr** [comment]
> Next week. Let's do this. For realsies. We could stay late after work and drink beers or something.

:heart: :heart: :heart: :heart: :heart: :heart: 

Once finished, I'll sing the link from the mountain tops for all to hear. We will celebrate!

**Comment by mikebridge** [comment]
Speaking of [liquidmarkup.org](http://liquidmarkup.org), it doesn't seem like the custom filter  _prettyprint_ should be the first example people see, either on that site or in the [README](https://github.com/Shopify/liquid/blob/master/README.md).

```sh
 ~/work/liquid $ grep -r -i prettyprint *
example/server/templates/products.liquid:          {{product.description | prettyprint | paragraph }}
example/server/example_servlet.rb:  def prettyprint(text)
README.md:      {{ product.description | prettyprint | paragraph }}
```

**Comment by parkr** [comment]
So excited for this!

**Comment by admhlt** [comment]
> Speaking of liquidmarkup.org, it doesn't seem like the custom filter prettyprint should be the first example people see, either on that site or in the README.

I definitely agree, Mike. The new docs will have a (hopefully) much better introduction with simpler and better-presented examples.

**Comment by admhlt** [comment]
We made some progress, @parkr, but we're not done yet. We're hoping to hit this again next week. Stay tuned.

**Comment by parkr** [comment]
> We made some progress, @parkr, but we're not done yet. We're hoping to hit this again next week. Stay tuned.

@admhlt Thank you so much! Really looking forward to this. It'll be a huge help to all those using Liquid. :)

**Comment by parkr** [comment]
_hopeful coati_

hopeful coati

Can I help with anything?

**Comment by parkr** [comment]
@admhlt @Tetsuro, Any updates here? Would _love_ to see this out before the beginning of Feb!

**Comment by Tetsuro** [comment]
@parkr!!  We're currently putting getting more people to get this done. We want to work toward a more automated documentation for Liquid, much like we do with our current API docs. 

In the meantime I'm doing an audit of what's left to do, so that we can at least ship the static content that we have now. I think we're actually pretty close in terms of all the content being there.

**Comment by admhlt** [comment]
@Tetsuro and I are getting together again next Friday (Jan 29) to get more work done on this. So close. :droplet: :sweat_drops: :droplet:

**Comment by parkr** [comment]
> @Tetsuro and I are getting together again next Friday (Jan 29) to get more work done on this. So close.   

I am reallllllllly excited for this to :ship:. I keep wanting to link out to these docs to help Jekyll users! :soon:

**Comment by parkr** [comment]
@admhlt @Tetsuro Yay! Thanks for working on this last Friday. Looking good to ship? If not, what can be done later and what needs to be done before shipping? Maybe there's a way we can get a minimal version out so we can start referencing, then we can flesh out later with examples and so forth. :smile:

**Comment by Tetsuro** [comment]
Heya @park

**Comment by Tetsuro** [comment]
Heya @parkr! We made some good progress last Friday, and we're getting really close to shipping. The last thing we need to do is to figure out what we want to show on the home page, and make sure all the content under the "Basics" and "Tags" section are :ok_hand:. We're pretty happy with where the "Filters" section is now. 

Once we do one last push to wrap up the "Basics" and "Tags" sections, can we ping you to have you do a quick lookover of the content? 

Finally, if you have any suggestions for the front end side of things, please let me know!

**Comment by parkr** [comment]
Thanks for the update, @Tetsuro :heart: Just pulled it down and ran `npm install grunt-cli && node_modules/.bin/grunt`. Site looks great :smile: :smile: 

> The last thing we need to do is to figure out what we want to show on the home page

[liquidmarkup.org](http://liquidmarkup.org/) currently shows a snippet of Liquid and briefly describes its purpose. I like that. What you have now is great, too! Perhaps tobi has a preference?

> Once we do one last push to wrap up the "Basics" and "Tags" sections, can we ping you to have you do a quick lookover of the content?

Absolutely!!!

**Comment by pushrax** [comment]
I like the circular dependency introduced by using Jekyll in Liquid.

**Comment by parkr** [comment]
> I like the circular dependency introduced by using Jekyll in Liquid.

:laughing: 

@Tetsuro @admhlt Need some :eyes: ?

**Comment by Tetsuro** [comment]
Hey @parkr! Is there anything on the installation or Rails side that you think we should add to the docs? 

I'm going to spend some time later this week to (hopefully) finish the home page, and do a final review of the "Basics" and "Tags" sections.

**Comment by parkr** [comment]
> Is there anything on the installation or Rails side that you think we should add to the docs?

I think `gem install liquid` is all that's really needed, but an "Integrations" page would make sense. It's kind of like your "Liquid for Developers" page, where you describe how to render a tag, register a tag, etc. I don't think that's as important here in this first iteration, though.

I am _REALLY REALLY REALLY_ excited for this to ship!

**Comment by admhlt** [comment]
> I am REALLY REALLY REALLY excited for this to ship!

Me too!

**Comment by parkr** [comment]
@admhlt @Tetsuro Helllooooooooo! Can we ship this as-is and iterate later? Every day, going through issues, I wish I could link out to your documentation, but am reminded of this PR instead. We can iterate later :smile:

**Comment by Tetsuro** [comment]
@parkr I worked some more on this some night ago, and again tonight. Updates include:

- Simpler `Basics > Introduction` 
- Nicer homepage 
- Update styles for code examples

**Todo, but can be included after**

- Consider responsive design (hide sidebar on small screens)
- Better a11y for screen readers
- Better documentation for developers. Installation, how to use Liquid, more backend-y stuff...  I will defer this to more rails-savvy folks. 

@admhlt @parkr @fw42 can I get a proofread on the docs to make sure its A) in a shippable state and B) language is consistent? Thanks!!! 

Also, I have a lot of git squashin' to do here... LOL

**Comment by admhlt** [comment]
Proofreading now! I'll probably push a few minor changes this morning.

**Comment by admhlt** [comment]
Good stuff! I've fixed a few things. Here's a few things I've noticed are missing, but aren't mission-critical:
- we don't list the methods available for `forloop` objects
- we should link to the Shopify Liquid docs somewhere, and explain the difference between flavours
- we should explain cycle groups better
- it would be nice to have a top-level page explaining filters (and maybe listing all filter pages and their contents) – specifying that you can string together filters and whatnot

**Comment by Tetsuro** [comment]
Thanks for taking a look @admhlt! Great feedback as well. Let's make Issues for those so that we can fix them post- :ship:.

**Comment by parkr** [comment]
Hooray! Well done both of you. :smile: 

- I had to move `_sass/main.scss` to `css/main.scss` and add empty YAML front matter to get it to compile. If that's not the way to do it, then the `README.md` file need to be updated!
- On `/tags/control-flow/`, it'd be nice to have some explanation of what operators function in the `if`, and `unless` block, like if I can write `{% if page.tags is empty %}` or not. If  only `>`, `<`, `>=`, `<=`, `==`, and `!=` work, then we should share that.
- On that same page, where you have `<!-- If customer.name = 'anonymous' -->`, it might be good to say **When** instead of **If**, so you'd get `<!-- When customer.name = 'anonymous' -->`. It explains the output a bit better
- Could we linkify the headers? Either as `a` tags which link to themselves or something fancier like what we do on jekyllrb.com (e.g. go [here](http://jekyllrb.com/docs/templates/) and hover over any h2 or lower in the document). It would make it easier to share specific content :smile: 
- Do you have a design philosophy for Liquid? If @tobi has a paragraph or two that explains the rationale for Liquid, why you'd use it, why you wouldn't, and what its project bounds are, that might be useful to include. Perhaps it could go on the home page?

Everything else looks great!!! :tada: :clap: :pray: :bow:

**Comment by parkr** [comment]
Merge by accident? Let it haaapppeennnnn :smile: 

hooray!!!!!!

**Comment by Tetsuro** [comment]
Lol @parkr we're super-close. I did indeed merge but ran into some CSS issues... still aiming to merge today though!

I looked over your feedback and they're all great - thanks for that! We're going to go ahead and merge just so we can get something out, and then we'll work on going through the feedback. 

Thanks :)

**Comment by fw42** [comment]
https://shopify.github.io/liquid/

**Comment by tobi** [comment]
Pretty broken on mobile :-(
On Fri, Mar 18, 2016 at 6:08 PM Florian Weingarten <notifications@github.com>
wrote:

> https://shopify.github.io/liquid/
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Shopify/liquid/pull/513#issuecomment-198562381>
>
-- 
- tobi
CEO Shopify

**Comment by Tetsuro** [comment]
@tobi fix on the way.

**Comment by tobi** [comment]
📱😍
On Fri, Mar 18, 2016 at 6:26 PM tetchi <notifications@github.com> wrote:

> @tobi <https://github.com/tobi> fix on the way.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Shopify/liquid/pull/513#issuecomment-198565333>
>
-- 
- tobi
CEO Shopify
